src/animations.ts

```diff
@@ -111,105 +455,107 @@ let FAILED_KARAOKE_UPDATE_ATTEMPTS = 0
-            const element = document.getElementById(step.id)
-            if (animationTime >= step.start && animationTime < step.end) {
-                element.style.opacity = '1'
-            } else {
-                element.style.opacity = '0'
-            }
-        }
+// Set the animation back to the initial state.
+function sampleQuestionAnimationCleanup(animationName: AnimationName) {
+    const animation = ANIMATIONS[animationName]
+    setRadialProgressBar(animation, 0)
+    attemptUpdateKaraoke(animation.karaoke, 0, null)
+}
 
-        setRadialProgressBar(animation, animationTime)
-        attemptUpdateKaraoke(animation.karaoke, animationTime)
 
-        window.requestAnimationFrame(yourLittleOneAnimation)
-    }
-}
+let FAILED_GET_SAMPLE_QUESTION_COMPONENTS_ATTEMPTS = 0
+
+// Since the sample questions are created programmatically from the CMS, they may not be in the DOM by the time this
+// script runs (even with `defer`). Attempt to fetch the components for 10 seconds before sending a Sentry alert.
+function getSampleQuestionComponents() {
+    const sampleQuestionsSlider = $('.sample-questions-slider')
+    for (const animationName of SAMPLE_QUESTION_ANIMATIONS) {
+        // The components for these animations are programmatically created from the CMS collection, so they cannot be
+        // reliably found with a class or id. Use the audio URL to find the slide associated with this animation.
+        const slide = sampleQuestionsSlider.find(
+            `div[data-element='url']:contains('${ANIMATIONS[animationName].expectedAudioSrc}')`
+        ).closest('.sample-questions-slide')
 
+        if (slide.length) {
+            ANIMATIONS[animationName].progressBar = $(slide).find('.sample-question_button-progress circle')[0] as unknown as SVGCircleElement
+            ANIMATIONS[animationName].karaoke.speakerElements[1].container = slide.find('.sample-question_quote-container')[0]
+            ANIMATIONS[animationName].karaoke.speakerElements[1].quote = slide.find('.sample-question_quote')[0]
 
-// Trigger the animation to start when the play button is scrolled into view.
-function onPlayButtonIntersection(entries: IntersectionObserverEntry[]) {
-    const playButtonEntry = entries[0]
-    const animationName = (playButtonEntry.target as HTMLElement).dataset['animation']
-    if (playButtonEntry.isIntersecting && animationName in ANIMATIONS) {
-        // Start a new animation when it scrolls into view.
-        CURRENT_ANIMATION_INFO.karaokeState = null
-        CURRENT_ANIMATION_INFO.name = animationName
-        let timeOffset = 0
-        if (PLAYER.querySelector('source').src === ANIMATIONS[animationName].expectedAudioSrc && !PLAYER.paused) {
-            timeOffset = PLAYER.currentTime * 1000
+            // Set the initial state of the animation.
+            ANIMATIONS[animationName].cleanupAnimation()
+        } else if (FAILED_GET_SAMPLE_QUESTION_COMPONENTS_ATTEMPTS >= 10) {
+            Sentry.captureMessage(
```

Comment:
It might be linked with my ad blocker (Brave) but Sentry is not guaranteed to be properly defined, with the pure JS integration, based on whether or not the script has been loaded. You might want to wrap the call to Sentry in a safer helper (to make sure not everything crashes after the call to Sentry, in that scope)

Reply:
![image](https://user-images.githubusercontent.com/9143117/187772982-4a60413a-055a-48bf-9d3f-9f530672f06f.png)

Reply:
Ahh thanks for raising this, great catch! Yes we can improve to safely call Sentry

Reply:
Interesting, Brave is also blocking our analytics script from this repo. Could the filename by itself be enough to cause Brave to block this?
<img width="597" alt="Screen Shot 2022-08-31 at 6 26 29 PM" src="https://user-images.githubusercontent.com/68879394/187796074-37dc3739-b089-41bb-abeb-93ea4b7954bd.png">